src/CGCOMP.CPP

```diff
@@ -834,6 +834,163 @@ void ANDAT::ad_DupHERV( const ANDAT& ad)		// dup HERV values
 	ad_pFan = ad.ad_pFan;
 }	// ANDAT::ad_DupHERV
 //===============================================================================
+RC DOAS::oa_CkfDOAS()	// input checks
+{
+	RC rc = RCOK;
+	return rc;
+}
+//-----------------------------------------------------------------------------
+RC DOAS::oa_Setup(DOAS* iRat)
+{
+	RC rc = RCOK;
+	*this = *iRat;		// copy record, incl name, excl internal front overhead.
+
+	rc |= oa_CkfDOAS();		// check run record
+
+	// Calculate total design flow rates from child IXFERS
+	float supFanVfDemand = 0.f;
+	float exhFanVfDemand = 0.f;
+
+	IZXRAT* izx;
+	RLUP( IzxR, izx)
+	{	
+		if (izx->iz_IsDOAS() && izx->iz_doas == ss)
+		{
+			if (izx->iz_vfMin > 0)
+			{
+				supFanVfDemand += izx->iz_vfMin;
+			}
+			else
+			{
+				exhFanVfDemand -= izx->iz_vfMin;
+			}
+		}	
+	}
+
+	// If fan design flows are not set, set them from IZXFER flows
+	//   Note: This may not be reliable if iz_vfMin is an expression
+	// Warn if vfDs is not large enough to meet IZXFER demand
+	bool supFanVfDsIsSet = IsSet(DOAS_SUPFAN + FAN_VFDS);
+	bool exhFanVfDsIsSet = IsSet(DOAS_EXHFAN + FAN_VFDS);
+
+	if (!supFanVfDsIsSet)
+	{
+		oa_supFan.vfDs = supFanVfDemand;
+	}
+	else
+	{
+		float supFlowRatio = oa_supFan.vfDs / supFanVfDemand;
+		if(supFlowRatio < 0.99f)
+		{
+			oWarn( "Sum of IZXFER supply flows referencing DOAS is larger than vfDs.\n"
+				"    IZXFER supply flows will be reduced proportionately"
+			);
+
+			// Adjust IZXFER flows
+			IZXRAT* izx;
+			RLUP( IzxR, izx)
+			{	
+				if (izx->iz_IsDOAS() && izx->iz_doas == ss)
+				{
+					if (izx->iz_vfMin > 0)
+					{
+						izx->iz_vfMin *= supFlowRatio;
+					}
+				}	
+			}
+		}
```

Comment:
This can maybe be turned into a function to eliminate repetition on the Exhaust side.